### PR TITLE
fix: add check for skip authz to all authz check calls

### PR DESCRIPTION
**Pull request by Vic-Dev**
Adds check for skip authz flag to authz check calls

## Description

All authz check calls should allow for skipping authz when the skip authz flag is set.

## Review Checklist
- [x] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [x] I have added tests to validate that the change in functionality is working as expected

If you haven't done so yet, we would appreciate it if you could star the [OpenFGA repository](https://github.com/openfga/openfga). :)

### CODE REVIEW

File: pkg/server/server_authz_test.go [hunk 1]
```diff
@@ -700,6 +707,80 @@ func TestCheckAuthz(t *testing.T) {
...
 }
 
+func TestGetAccessibleStores(t *testing.T) {
+	t.Cleanup(func() {
+		goleak.VerifyNone(t)
+	})
+	ds := memory.New()
+	t.Cleanup(ds.Close)
+
+	t.Run("check_no_authz", func(t *testing.T) {
```
Comment (adriantam) [inline pkg/server/server_authz_test.go:717]: nitpik: check_no_authz reminds me of "check" as in OpenFGA.  Perhaps avoid the word "check"?

File: pkg/server/server_authz_test.go [hunk 2]
```diff
@@ -700,6 +707,80 @@ func TestCheckAuthz(t *testing.T) {
...
+			ctx := authclaims.ContextWithSkipAuthzCheck(context.Background(), true)
+
+			_, err := openfga.getAccessibleStores(ctx)
+			require.NoError(t, err)
+		})
+
+		t.Run("error_with_no_client_id_found", func(t *testing.T) {
+			_, err := openfga.getAccessibleStores(context.Background())
+			require.Error(t, err)
+			require.Equal(t, "rpc error: code = InvalidArgument desc = client ID not found in context", err.Error())
```
Comment (adriantam) [inline pkg/server/server_authz_test.go:748]: Will it be better to use `require.ErrorIs(t, fmt.Errorf("rpc error: code = InvalidArgument desc = client ID not found in context", err)`?

File: pkg/server/server.go [hunk 3]
```diff
@@ -1618,24 +1625,29 @@ func (s *Server) checkCreateStoreAuthz(ctx context.Context) error {
...
-		}
+	if authclaims.SkipAuthzCheckFromContext(ctx) {
+		return nil, nil
+	}
 
-		list, err := s.authorizer.ListAuthorizedStores(ctx)
-		if err != nil {
-			return nil, err
-		}
+	err := s.authorizer.AuthorizeListStores(ctx)
```
Comment (rhamzeh) [inline pkg/server/server.go:1632]: I'm assuming we removed the `if s.authorizer != nil` b/c it can never be nil now right? It's noop by default?

Comment (Vic-Dev) [reply pkg/server/server.go:1632]: Yup there's a noop function for both of these where it just returns nil

**Review by jpadilla — DISMISSED** [review]

**Review by rhamzeh — APPROVED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -19,6 +19,7 @@ Try to keep listed changes to a concise bulleted list of simple explanations of
 ### Fixed
 * Label ListUsers API calls [#2000](https://github.com/openfga/openfga/pull/2000)
 * Fixed incorrect dispatch counts in List Objects [2013](https://github.com/openfga/openfga/pull/2013)
+* Fixed a bug in access control where write, create store, and list stores were not skippable [#2033](https://github.com/openfga/openfga/pull/2033)
```
Comment (rhamzeh) [inline CHANGELOG.md:22]: We don't need a changelog change for this PR considering the original feature hasn't been released yet

**Review by adriantam — APPROVED** [review]

**Review by rhamzeh — APPROVED** [review]

**Pull request merged**